Library/Homebrew/compat/md5.rb

```diff
@@ -1,26 +1,35 @@
 class Formula
   def self.md5(val)
+    # Don't throw an error at load time, just install time.
     stable.md5(val)
   end
 end
 
 class SoftwareSpec
   def md5(val)
+    odie <<-EOS.undent
+      Using MD5 has been dropped for security reasons.
+      Please switch to SHA-256.
+    EOS
```

Comment:
> What makes it worse is that `odie` unlike `raise RuntimeError` won't be properly caught. One of examples is `brew tap` will just exit without untaping the tap if one of formulae happens to use such DSL.

This is IMO a very strong argument for at least “downgrading” to an exception instead of killing the `brew` process, so that the deprecation error can be properly caught by `brew readall` (as part of `brew tap`) and proper cleanup can happen when trying to install a tap that still uses the deprecated DSL.

I don't think anybody in here is campaigning for keeping MD5 support alive in the sense that we allow formulae that rely on that to be successfully loaded (or even installed).